probe.c

```diff
@@ -61,120 +69,161 @@ void probe_init()
 
 void probe_check()
 {
-  if (probe_get_active_sensor_state())
+  if (probe_get_active_sensor_state()) {
+    // Stop looking for probe
     probe.isprobing = 0;
-}
-
-void probe_plan_move(uint8_t sensor, uint8_t axis, uint8_t dir)
-{
-  /*  sensor - Which sensor to home
-   *  axis - Which axis to move to the sensor
-   *  dir - true = neg; false = pos
-   */
-
-  // Set the active probe
-  probe.active_sensor = sensor;
-
-  // Approach has all bits set (neg dir) or none (pos dir)
-  const uint8_t approach = dir ? ~0 : 0;
-
-  // Limit travel distance to the max flag
-  const float travel = settings.max_travel[axis];
-  const uint8_t flipped = settings.homing_dir_mask >> X_DIRECTION_BIT;  //assumes keyme configuration.
- 
-  // Set target for axis
-  float target[N_AXIS] = {0};
-  target[axis] = ((flipped & (1 << axis)) ^ approach) ? -travel : travel;
-
-  // Set speed
-  //TODO: Allow for this rate to be set over serial / passed as a variable
-  const float probe_rate = settings.homing_seek_rate[axis];
-
-  // Reset plan and stepper buffers
-  plan_reset();
-  st_reset();
-
-  // Plan moition. Planner buffer should be empty, bypass mc_line() 
-  plan_buffer_line(target, probe_rate, false, LINENUMBER_EMPTY_BLOCK); 
-
-  // Tell the system we are probing
-  probe.isprobing = 1;
-
-  // Enable hard limit checking
-  limits_enable(LIMIT_MASK & HARDSTOP_MASK, 0);
 
+    // Report the position where the probe was found
+    //TODO report_probe_parameters() Maybe rather in probe_loop or probe_move_to_sensor
+  }
 
+  // Check for ESTOP
+  if (ESTOP_PIN & ESTOP_MASK) {
+    sys.alarm |= ALARM_ESTOP;
+    SYS_EXEC |= (EXEC_FEED_HOLD | EXEC_ALARM | EXEC_CRIT_EVENT);
+  }
 }
 
-void probe_loop()
+uint8_t probe_loop()
```

Comment:
What is the benefit? bool will take the same space (one byte) as uint8_t. From the docs:

> bool
>         Expands to _Bool.
> true
>         Expands to the integer constant 1.
> false
>         Expands to the integer constant 0.
>     __bool_true_false_are_defined
>     Expands to the integer constant 1.